### PR TITLE
trainers swiper

### PR DESCRIPTION
**Pull request by AhmedMughessib**

### CODE REVIEW

**Review by HamdyPs — APPROVED** [review]

File: src/pages/user/home/index.jsx [hunk 1]
```diff
@@ -1,32 +1,45 @@
...
-        <Box mb={10} position="relative">
-          <SwiperComp images={images} />
+  const handleDataChange = (newData) => {
+    setTrainersData(newData);
+  };
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const trainers = await axios.get('http://localhost:5050/api/users/trainers');
+        await handleDataChange(trainers.data.allTrainers);
```
Comment (AdhamHaisami) [inline src/pages/user/home/index.jsx:19]: setState directly. no need for a raper function, and why do you use await in here?

File: src/pages/user/home/index.jsx [hunk 2]
```diff
@@ -1,32 +1,45 @@
...
-        </Box>
-        <Box mb={10} position="relative">
-          <SwiperComp images={images} />
+  const handleDataChange = (newData) => {
+    setTrainersData(newData);
+  };
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const trainers = await axios.get('http://localhost:5050/api/users/trainers');
```
Comment (AdhamHaisami) [inline src/pages/user/home/index.jsx:18]: we don't add the full URL, it's already declared in the vite config file. add only the endpoint

**Review by AdhamHaisami — CHANGES_REQUESTED** [review]

File: src/pages/user/home/index.jsx [hunk 3]
```diff
@@ -1,32 +1,45 @@
...
-          <SliderComp />
-        </Box>
-        <Box mb={10} position="relative">
-          <SwiperComp images={images} />
+  const handleDataChange = (newData) => {
+    setTrainersData(newData);
+  };
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
```
Comment (HamdyPs) [inline src/pages/user/home/index.jsx:17]: u don't need to use try catch, instead u need to check the status that comes from response

File: src/pages/user/home/index.jsx [hunk 4]
```diff
@@ -1,32 +1,45 @@
...
-        <Box mb={10}>
-          <SliderComp />
-        </Box>
-        <Box mb={10} position="relative">
-          <SwiperComp images={images} />
+  const handleDataChange = (newData) => {
+    setTrainersData(newData);
+  };
+  useEffect(() => {
+    const fetchData = async () => {
```
Comment (HamdyPs) [inline src/pages/user/home/index.jsx:16]: move this function out of useEffect

**Review by HamdyPs — CHANGES_REQUESTED** [review]

File: src/components/swiper/index.jsx [hunk 5]
```diff
@@ -23,13 +23,13 @@ const SwiperComp = ({ images }) => (
...
 
-      {images.map((item) => (
+      {trainers.map((item) => (
         <SwiperSlide style={{
           padding: '40px 0px', display: 'flex', flexDirection: 'column', gap: 15,
         }}
         >
-          <img height={350} width="300px" style={{ margin: 'auto', borderRadius: 40 }} src={item} alt={item} />
-          <Typography variant="h4" fontWeight={700} textAlign="center">trainer</Typography>
+          <img height={350} width="300px" style={{ margin: 'auto', borderRadius: 40 }} src={item.image} alt={item} />
```
Comment (HamdyPs) [inline src/components/swiper/index.jsx:31]: change alt value to item.anything from the object

**Review by HamdyPs — CHANGES_REQUESTED** [review]

**Pull request closed without merging**